jsapp/js/searches.es6

```diff
@@ -117,9 +118,22 @@ function SearchContext(opts={}) {
       }
     },
     // remove asset from all search store lists
-    removeAsset(asset) {
-      // only update things if given asset matches the current context types
-      if (this.state.defaultQueryFilterParams?.assetType.includes(asset.assetType)) {
+    removeAsset(assetOrUid, isNonOwner) {
+      // only update things if given asset matches the current context types or
+      // a non-owner removed their own permissions
+      if (isNonOwner || this.state.defaultQueryFilterParams?.assetType.includes(asset.assetType)) {
```

Comment:
@duvld I've found a bug here: `asset.assetType` doesn't exist here. It happend to me when I deleted an asset from My Library.